src/multio/action/scale/Scaling.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "eckit/config/LocalConfiguration.h"
+#include "multio/config/ComponentConfiguration.h"
+#include "multio/message/Glossary.h"
+#include "multio/message/Message.h"
+
+namespace multio::action {
+
+using message::glossary;
+
+class ScaleScaling {
+private:
+    bool hasScaling_;
+    std::map<std::string, double> scaleFactor_;
+
+public:
+    explicit ScaleScaling(const config::ComponentConfiguration& compConf);
+    double getScalingFactor(const std::string paramID) const;
+    template <typename Precision>
+    void applyScaling(message::Message& msg) const {
+        if (hasScaling_) {
+            std::ostringstream key;
+            std::string cparam{"xxx"};
+
+            // Extract the scaling parameter
+            if (auto param = msg.metadata().getOpt<std::string>(glossary().param); param) {
+                cparam = *param;
+            }
+            else if (auto paramId = msg.metadata().getOpt<std::int64_t>(glossary().paramId); paramId) {
+                cparam = std::to_string(*paramId);
+            }
+            else {
+                throw eckit::SeriousBug{"param/paramId metadata not present", Here()};
+            }
+
+            // Find the scaling factor
+            double scaleFactor = getScalingFactor(cparam);
+
+            // Ensure the payload size is compatible with Precision type
+            size_t size = msg.payload().size() / sizeof(Precision);
+            if (size == 0) {
+                throw eckit::SeriousBug{" Payload is empty: Scaling Action: " + msg.metadata().toString(), Here()};
+            }
+
+            // Access the payload data
+            auto data = static_cast<Precision*>(msg.payload().modifyData());
+            if (!data) {
+                throw eckit::SeriousBug{
+                    " Payload data could not be modified: Scaling Action: " + msg.metadata().toString(), Here()};
+                std::cout << "Error: Payload data could not be modified." << std::endl;
+                return;
+            }
+            // Apply the scaling factor using std::transform
+            std::transform(data, data + size, data,
```

Comment:
In this case, `std::for_each` is sufficient. It is simpler and thus preferable, I'd say.

Reply:
I think I disagree in this case, I'm sorry. I think transform makes it clearer as to the context of the underlying action (which in fact is a transformation of data in place). I agree it could be done with for_each but I don't think it would be simpler. Nevertheless I'm also not very partial on this line of code. So if you really prefer for_each I can change the line to 
`std::for_each(data, data + size, [scaleFactor](Precision& value) {
    value = static_cast<Precision>(value * scaleFactor);
});`